pennylane_lightning/_serialize.py

```diff
@@ -58,162 +40,210 @@
-        inverses, and a list of matrices for the operations that do not have a dedicated kernel.
-    """
-    names = []
-    params = []
-    wires = []
-    mats = []
-
-    uses_stateprep = False
-
-    for o in tape.operations:
-        if isinstance(o, (BasisState, QubitStateVector)):
-            uses_stateprep = True
-            continue
-        elif isinstance(o, Rot):
-            op_list = o.expand().operations
+    def _hamiltonian(self, ob, wires_map: dict, use_csingle: bool):
+        if use_csingle:
+            rtype = np.float32
+            hamiltonian_obs = self.HamiltonianC64
+        else:
+            rtype = np.float64
+            hamiltonian_obs = self.HamiltonianC128
+
+        coeffs = np.array(unwrap(ob.coeffs)).astype(rtype)
+        terms = [self._ob(t, wires_map, use_csingle) for t in ob.ops]
+        return hamiltonian_obs(coeffs, terms)
+
+    def _pauli_word(self, ob, wires_map: dict, use_csingle: bool):
+        """Serialize a :class:`pennylane.pauli.PauliWord` into a Named or Tensor observable."""
+        if use_csingle:
+            named_obs = self.NamedObsC64
+            tensor_obs = self.TensorProdObsC64
         else:
-            op_list = [o]
+            named_obs = self.NamedObsC128
+            tensor_obs = self.TensorProdObsC128
+
+        if len(ob) == 1:
+            wire, pauli = list(ob.items())[0]
+            return named_obs(pauli_name_map[pauli], [wires_map[wire]])
+
+        return tensor_obs(
+            [named_obs(pauli_name_map[pauli], [wires_map[wire]]) for wire, pauli in ob.items()]
+        )
+
+    def _pauli_sentence(self, ob, wires_map: dict, use_csingle: bool):
+        """Serialize a :class:`pennylane.pauli.PauliSentence` into a Hamiltonian."""
+        if use_csingle:
+            rtype = np.float32
+            hamiltonian_obs = self.HamiltonianC64
+        else:
+            rtype = np.float64
+            hamiltonian_obs = self.HamiltonianC128
+
+        pwords, coeffs = zip(*ob.items())
+        terms = [self._pauli_word(pw, wires_map, use_csingle) for pw in pwords]
+        coeffs = np.array(coeffs).astype(rtype)
+        return hamiltonian_obs(coeffs, terms)
+
+    def _ob(self, ob, wires_map, use_csingle):
+        """Serialize a :class:`pennylane.operation.Observable` into an Observable."""
+        if isinstance(ob, Tensor):
+            return self._tensor_ob(ob, wires_map, use_csingle)
+        if ob.name == "Hamiltonian":
+            return self._hamiltonian(ob, wires_map, use_csingle)
+        if isinstance(ob, (PauliX, PauliY, PauliZ, Identity, Hadamard)):
+            return self._named_obs(ob, wires_map, use_csingle)
+        if ob._pauli_rep is not None:
+            return self._pauli_sentence(ob._pauli_rep, wires_map, use_csingle)
+        return self._hermitian_ob(ob, wires_map, use_csingle)
+
+    def _observables(self, tape: QuantumTape, wires_map: dict, use_csingle: bool = False) -> List:
```

Comment:
```suggestion
    def serialize_observables(self, tape: QuantumTape, wires_map: dict, use_csingle: bool = False) -> List:
```

I think since this is the explicit public interface, we can name it as above.